Modules/Quotas/test/Quotas.Domain.Tests/Tests/Tiers/TierTests.cs

```diff
@@ -47,6 +59,21 @@ public void Can_delete_quota_on_tier()
         tier.Quotas.Should().HaveCount(0);
     }
 
+    [Fact]
+    public void Cannot_delete_quota_on_queued_for_deletion_tier()
+    {
+        // Arrange
+        Tier.QUEUED_FOR_DELETION.Quotas.Add(new TierQuotaDefinition(MetricKey.NumberOfSentMessages, 0, QuotaPeriod.Total));
```

Comment:
Wait. Why is it possible to add something to the list of Quotas? It should at least be a ReadonlyCollection.

To add quota for the sake of this test, use the public API. Just like a user would.

Reply:
And still, it's somehow possible to add Quotas. ;) You just implemented it, so you should know how. :)